### DISCUSSION TITLE
Getting $primevue undefined error when trying to extend Dialog, Calendar component while creating new library

### DISCUSSION THREAD

**Discussion by mbaljeetsingh** [primefaces/community#876] [PrimeVue]
I'm trying to build a library using the following guide
https://tailwind.primevue.org/guides/building-ui-library/

Tried to export basic components like buttons, InputText, etc. It works fine.

But, I'm getting the following error when I click the show dialog. It works fine in the library project.

![image](https://github.com/primefaces/community/assets/872762/49731d74-bace-46d6-be00-2e09370a4397)

For more context, I have one project named `test-library` which is making use of `primevue tailwind` components. I'm exposing the new components.

Then, I'm trying to access them in the new vue 3 project.

Another issue I'm having is with exposing the calendar component. When I try to use it, I get the following error.  

Again, it works fine in the `test-library` project.

> The issue is related to $primevue. I think It doesn't have access to it anymore. 

![image](https://github.com/primefaces/community/assets/872762/825e0a25-b358-47f4-81ed-fea37f802917)
![image](https://github.com/primefaces/community/assets/872762/27fc8709-7443-41c3-bb11-2b895699e214)

**Comment by cagataycivici** [comment]
Are you using Vite or Nuxt in your setup? The plugin seems to be missing, please see;

**For Vite**
https://primevue.org/vite/#plugin

**For Nuxt**
https://primevue.org/nuxt/#plugin

Also here are two examples;

**Vite**
https://github.com/primefaces/primevue-examples/tree/main/vite-unstyled-tailwind

**Nuxt**
https://github.com/primefaces/primevue-examples/tree/main/nuxt-unstyled-tailwind

**Comment by mbaljeetsingh** [comment]
@cagataycivici 

I'm using vite. As I said earlier, I'm trying to build a library on top of primevue. So, If I have to still primevue as a dependency to consume the new library, I'm not sure it is a good experience.

In library project, I'm using it like following,
<img width="1375" alt="image" src="https://github.com/primefaces/community/assets/872762/9ca3b43a-fc64-4dea-b9ed-d8e82ec343e2">

and this is how I'm consuming in new vue project,
<img width="1038" alt="image" src="https://github.com/primefaces/community/assets/872762/28e1e6f3-4113-4271-8deb-9acb7f276e27">

I'm currently testing with npm link locally. Let me know If there is a way to use Primevue to create a library. I would want to keep Primevue dependency in the library project only. 

Let me know if you any more details from me. 

Thanks,
Baljeet

**Comment by mbaljeetsingh** [comment]
Hi @cagataycivici 

Is there any potential solution?

**Reply by mbaljeetsingh** [reply]
Hi

I managed to fix it by exporting `export Primevue  from 'primevue/config';` in the library project and then using `app.use(Primevue)`. Let me know if this is the correct approach.

Thanks
